docopt_private.h

```diff
@@ -287,14 +287,14 @@ namespace docopt {
 
 	class Required : public BranchPattern {
 	public:
-		using BranchPattern::BranchPattern;
+		using BranchPattern::&const BranchPattern;
```

Comment:
Yikes, what is this lol ... This cant be legal code? Even if it is, this is not better than what was there before.